lib/features/email/data/datasource_impl/email_hive_cache_datasource_impl.dart

```diff
@@ -149,4 +152,37 @@ class EmailHiveCacheDataSourceImpl extends EmailDataSource {
       return await _emailCacheManager.storeEmail(accountId, session.username, email.toEmailCache());
     }).catchError(_exceptionThrower.throwException);
   }
+
+  @override
+  Future<void> storeOpenedEmailToCache(Session session, AccountId accountId, DetailedEmail detailedEmail) {
+    return Future.sync(() async {
+      final task = HiveTask(
+          runnable: () async {
+
+            final isDuplicateDetailedEmail = await _detailedEmailCacheManager.checkDuplicateOpenedEmail(accountId, session.username,detailedEmail);
+            log('EmailHiveCacheDataSourceImpl::storeOpenedEmailToCache():isDuplicateDetailedEmail: $isDuplicateDetailedEmail');
+
+            if (isDuplicateDetailedEmail) {
+              return Future.value();
+            }
+
+            final fileSaved = await _fileUtils.saveToFile(
+              nameFile: detailedEmail.emailId.asString,
+              content: detailedEmail.htmlEmailContent ?? '',
+              folderPath: detailedEmail.folderPath,
+              extensionFile: ExtensionType.text.value
+            );
+
+            log('EmailHiveCacheDataSourceImpl::storeOpenedEmailToCache():fileSavedPath: ${fileSaved.path}');
+
+            final detailedEmailSaved = detailedEmail.fromEmailContentPath(fileSaved.path);
+
+            await _detailedEmailCacheManager.storeOpenedEmail(
+              accountId,
+              session.username,
+              detailedEmailSaved);
+          });
+      return await _openedEmailCacheWorkerQueue.addTask(task);
```

Comment:
I think we dont need to `await` for `addTask`. WDYT?
cc: @dab246

Reply:
Should remove `await`